server/src/__tests__/linter.test.ts

```diff
@@ -0,0 +1,154 @@
+import * as path from 'path'
+import * as LSP from 'vscode-languageserver'
+
+import { FIXTURE_DOCUMENT, FIXTURE_FOLDER } from '../../../testing/fixtures'
+import Linter, { assertShellcheckResult } from '../linter'
+
+function textToDoc(txt: string) {
+  return LSP.TextDocument.create('foo', 'bar', 0, txt)
+}
+
+describe('linter', () => {
+  it('should set canLint to false if executable empty', () => {
+    expect(new Linter({ executablePath: null }).canLint).toBe(false)
+  })
+
+  it('should set canLint to true if executable not empty', () => {
+    expect(new Linter({ executablePath: null }).canLint).toBe(false)
+  })
+
+  it('should set canLint to false when linting fails', async () => {
+    jest.spyOn(console, 'error').mockImplementation(() => {}) // eslint-disable-line
+    try {
+      const linter = new Linter({
+        executablePath: '77b4d3f6-c87a-11ec-9b62-a3c90f66d29f',
+      })
+      expect(await linter.lint(textToDoc(''), [])).toEqual([])
+      expect(linter.canLint).toBe(false)
+    } finally {
+      ;(console.error as any).mockRestore()
+    }
```

Comment:
This can be simplified a bit as `mockImplementation` doesn't need to take a function and mocks are automatically cleaned up. Besides adding a small assertion on the log would be nice.

```suggestion
    jest.spyOn(console, 'error').mockImplementation()
    const executablePath = '77b4d3f6-c87a-11ec-9b62-a3c90f66d29f'
    const linter = new Linter({
      executablePath,
    })
    expect(await linter.lint(textToDoc(''), [])).toEqual([])
    expect(linter.canLint).toBe(false)
```